src/vs/workbench/contrib/notebook/browser/controller/editActions.ts

```diff
@@ -621,3 +625,62 @@ registerAction2(class SelectNotebookIndentation extends NotebookAction {
 		return;
 	}
 });
+
+registerAction2(class CommentSelectedCellsAction extends NotebookMultiCellAction {
+	constructor() {
+		super({
+			id: COMMENT_SELECTED_CELLS_ID,
+			title: localize('commentSelectedCells', "Comment Selected Cells"),
+			keybinding: {
+				when: ContextKeyExpr.and(
+					NOTEBOOK_EDITOR_FOCUSED,
+					NOTEBOOK_CELL_LIST_FOCUSED,
+					ContextKeyExpr.not(InputFocusedContextKey),
+					NOTEBOOK_EDITOR_EDITABLE,
+					NOTEBOOK_CELL_EDITABLE
+				),
+				primary: KeyMod.CtrlCmd | KeyCode.Slash,
+				weight: KeybindingWeight.WorkbenchContrib
+			}
+		});
+	}
+
+	async runWithContext(accessor: ServicesAccessor, context: INotebookCommandContext): Promise<void> {
+		const languageConfigurationService = accessor.get(ILanguageConfigurationService);
+
+		const selectedCellEditors: ICodeEditor[] = [];
+		context.selectedCells.forEach(cell => {
+			const findContext = { notebookEditor: context.notebookEditor, cell };
+			const foundEditor = findTargetCellEditor(findContext, cell);
+			if (foundEditor) {
+				selectedCellEditors.push(foundEditor);
+			}
+		});
+
+		await Promise.all(selectedCellEditors.map(editor => {
```

Comment:
`Promise.all` seems redundant?

A review from someone with context in notebooks/editors would be more useful.

Reply:
the goal was to apply the edits in parallel across the relevant cells, though with nothing more relying on this, you're right it likely doesn't need the `promise.all`. Autopilot/habit from other work across async cell operations brought that in, will remove.